magento/meta-for-magento2
# Fix for issues in #72

### Description (*)

This PR fixes the issues outlined in Issue #72. It fixes the query-in-a-loop anti-pattern, the assumption that all items in a quote will be in a category (Which can cause a SQL error), the logic error where the category name array is overwritten for each item in the loop.

### Bug
* [75](https://github.com/magento/meta-for-magento2/issues/72) <Potential errors and performance issues in Tracker/InitiateCheckout>

### Fixed Issues (if relevant)
1. magento-commerce/facebook-for-magento2#<72>: Potential errors and performance issues in Tracker/InitiateCheckout


### Manual testing scenarios (*)

1. Create an item that has no categories, add it to the cart and check out. No error should occur. 

### Checklist
